File: backend-.NET/Vehiclecfg_group9/Vehiclecfg_group9/Models/InvoiceDetail.cs
using System;
using System.Collections.Generic;

namespace Vehiclecfg_group9.Models;

public partial class InvoiceDetail
{
    public int InvdtlId { get; set; }

    public double? DeltaPrice { get; set; }

    public int AltCompId { get; set; }

    public int CompId { get; set; }

    public long InvId { get; set; }

    public int ModelId { get; set; }

    public virtual ComponentMaster AltComp { get; set; } = null!;

    public virtual ComponentMaster Comp { get; set; } = null!;

    public virtual Invoice Inv { get; set; } = null!;

    public virtual ModelMaster Model { get; set; } = null!;
}

File: CDAC_PROJECT/JavaBackend/src/main/java/com/vehiclecfg/entities/InvoiceDetail.java
package com.vehiclecfg.entities;

import jakarta.persistence.*;

@Entity
@Table(name = "invoice_detail")
public class InvoiceDetail {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "invdtl_id")
    private Integer invdtlId;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "inv_id", nullable = false)
    private Invoice invoice;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "comp_id", nullable = false)
    private Component component;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "alt_comp_id", nullable = false)
    private Component alternateComponent;

    @Column(name = "delta_price")
    private Double deltaPrice;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "model_id", nullable = false)
    private Model model;

    public InvoiceDetail() {
    }

    public Integer getInvdtlId() {
        return invdtlId;
    }

    public void setInvdtlId(Integer invdtlId) {
        this.invdtlId = invdtlId;
    }

    public Invoice getInvoice() {
        return invoice;
    }

    public void setInvoice(Invoice invoice) {
        this.invoice = invoice;
    }

    public Component getComponent() {
        return component;
    }

    public void setComponent(Component component) {
        this.component = component;
    }

    public Component getAlternateComponent() {
        return alternateComponent;
    }

    public void setAlternateComponent(Component alternateComponent) {
        this.alternateComponent = alternateComponent;
    }

    public Double getDeltaPrice() {
        return deltaPrice;
    }

    public void setDeltaPrice(Double deltaPrice) {
        this.deltaPrice = deltaPrice;
    }

    public Model getModel() {
        return model;
    }

    public void setModel(Model model) {
        this.model = model;
    }

}